### PR TITLE
fix: conditional render makes wrapper lose focus

### PR DESCRIPTION
**Pull request by jrr997**
This is the referred [issue](https://github.com/ant-design/ant-design/issues/33917)

### CODE REVIEW

**Comment by afc163** [comment]
Would you mind to add a test case?

**Comment by jrr997** [comment]
finished test case

File: src/Dialog/index.tsx [hunk 1]
```diff
@@ -122,28 +125,35 @@ export default function Dialog(props: IDialogChildProps) {
...
-    }
-  }
-
+      // keep focus inside dialog
+      if (visible) {
+        if (e.keyCode === KeyCode.TAB) {
+          contentRef.current.changeActive(!e.shiftKey);
+        }
+      }
+    },
```
Comment (afc163) [inline src/Dialog/index.tsx:141]:
```suggestion
      if (visible && e.keyCode === KeyCode.TAB) {
        contentRef.current.changeActive(!e.shiftKey);
      }
    },
```

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/issues/33917#issuecomment-1295782475

**Pull request closed without merging**